### PR TITLE
FluxC: Fix post list progress bar getting stuck

### LINKED ISSUE #5135: FluxC: Post list refresh progress bar stuck loading
* Pull to refresh the post list and the progress bar will be visible
* Before the request finishes tap on the new post button
* Create a new post
* When you go back to the post list, both the new post progress bar and the refresh one will be visible
* After the request for new post finishes that progress bar will go away, but the refresh one will stay forever

double loading
forever

### PR DESCRIPTION
**Pull request by oguzkocer**
Fixes #5135. This PR fixes the mentioned issue with a reset however since we don't get the `onPostChanged` event, it's not possible to set the `mCanLoadMorePosts` flag. That shouldn't be a big deal though, once we do another fetch, it will be set to `false` if needed.

To test:
* Go into post list
* Trigger a manual fetch and immediately go into post detail
* Return back to post list and verify the progress bar is not visible anymore

### CODE REVIEW

**Comment by maxme** [comment]
So I just took a look on this and I think it makes more sense to move the dispatcher registration to onCreate on unregistration to onDestroy, like it was done in `SelectCategoriesActivity`.

I double checked and we need to do that in `PostsListFragment` but also in `CommentsListFragment`, these are the only places where this can happen.

**Comment by oguzkocer** [comment]
@maxime I've reverted the previous solution and moved the register/unregister to create/destroy actions. I've fixed the Comment one here as well as they are pretty much the same bug. We could update the issue title if it's a concern though.

File: WordPress/src/main/java/org/wordpress/android/ui/comments/CommentsListFragment.java [hunk 1]
```diff
@@ -104,17 +104,12 @@ public CommentStatus toCommentStatus() {
...
-        super.onStart();
         mDispatcher.register(this);
+        updateSiteOrFinishActivity(savedInstanceState);
     }
 
     @Override
-    public void onStop() {
+    public void onDestroy() {
         mDispatcher.unregister(this);
         super.onStop();
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/CommentsListFragment.java]: `super.onDestroy();` must be called here

Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/CommentsListFragment.java]: :facepalm: Sorry! Not good enough of an excuse, but I am practically sleep walking today :(

**Comment by maxme** [comment]
:shipit:

**Pull request merged**